### PR TITLE
Add gradients utility

### PR DESCRIPTION
**Pull request by mzlock**

### CODE REVIEW

File: plugins/gradients/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+plugins: [
+  require('@viget/tailwindcss-plugins/gradients'),
+],
+```
+
+The above configuration would create the following css, as well as their responsive variants:
+
+```css
+.gradient-90-blue {
+    background-image: linear-gradient(90deg, ${theme('colors.blue.400')} 50%, transparent);
```
Comment (gregkohn) [inline plugins/gradients/README.md]: I guess the color would also be computed to something? 🤷‍♂️small thing

File: plugins/gradients/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = () => ({ addUtilities, theme, variants }) => {
```
Comment (gregkohn) [inline plugins/gradients/index.js]: We just decided to use the new `plugin()` API that 1.2 introduced -- just means importing that fn and wrapping this export in it (can also skip the extra `() =>`). For now it doesnt do all that much -- mainly handles the invoking in `tailwind.config.js` automatically so no longer need to include `()` there -- but it allows some other features and feels like the "right" way to do it now

Example here: https://github.com/vigetlabs/tailwindcss-plugins/blob/master/plugins/rect/index.js

**Review by gregkohn — CHANGES_REQUESTED** [review]
tiny change but 👏

**Comment by mzlock** [comment]
@gregkohn @jeremyfrank - any preference or opinion on singular versus plural language for these?

- Classname: `.gradient-key` (singular)
- Config: `gradients: {}` (plural)
- Variants: `gradients: []` (plural)

File: plugins/gradients/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const plugin = require('tailwindcss/plugin')
+
+module.exports = plugin(({ addUtilities, e, theme, variants }) => {
+  const pluginConfig = theme('gradients', {})
+  const pluginVariants = variants('gradients', [])
+
+  const utilities = Object.keys(pluginConfig).map(([key, value]) => {
+    return {
+      [`.gradient-${key}`]: {
```
Comment (jeremyfrank) [inline plugins/gradients/index.js]: Might also want to use the `e` method to escape the class names, just incase someone tries to name the keys with fractions, e.g. `gradient-blue-1/2`. Actually I think the docs recommend always wrapping user provided selectors this way as a safeguard.

Comment (jeremyfrank) [reply plugins/gradients/index.js]: https://tailwindcss.com/docs/plugins/#escaping-class-names

**Review by jeremyfrank — APPROVED** [review]
Small tweak then 👍

**Comment by jeremyfrank** [comment]
Hmm, singluar vs plural, good question. Looking at the default config, maybe singular for all? Don't feel too strongly about it though.
https://github.com/tailwindcss/tailwindcss/blob/master/stubs/defaultConfig.stub.js

**Comment by gregkohn** [comment]
@mzlock good catch, I would definitely say to go with singular, that's convention as @jeremyfrank pointed out

**Comment by mzlock** [comment]
@gregkohn @jeremyfrank - Updated! What's the process for merging and getting changes on NPM? Should I let one of y'all merge and version things?

**Review by gregkohn — APPROVED** [review]

**Comment by gregkohn** [comment]
ah dang, one last thing @mzlock! can you add the gradients link to the main README?

**Comment by jeremyfrank** [comment]
Nice! No official process for when to put out a new version and publish to npm. Greg and I have just been handling that whenever we can.

**Pull request merged**